apps/prairielearn/src/pages/studentAssessmentExam/studentAssessmentExam.js

```diff
@@ -127,6 +128,7 @@ router.post(
       // students to create and start a new assessment instance.
       if (!checkPasswordOrRedirect(req, res)) return;
 
+      const client_fingerprint_id = await getClientFingerprintId(req, res);
```

Comment:
We did not want to use `clientFingerprints` middleware on the entire route. We only want to access the client fingerprint if, and only if, there is a post request to open the assessment. This prevents unwarranted fingerprints from viewing the assessment page before it has started.